test/functional/autocmd/tabnewentered_spec.lua

```diff
@@ -480,4 +480,40 @@ describe('tabpage/previous', function()
   --   does_not_switch_to_previous_after_entering_operator_pending('<C-W>g<Tab>'))
   it('does not switch to previous via <C-Tab> after entering operator pending',
     does_not_switch_to_previous_after_entering_operator_pending('<C-Tab>'))
+
+  local function does_not_switch_to_previous_when_editing_command_line(characters, completion_visible)
+    return function()
+      -- Add three tabs for a total of four
+      command('tabnew')
+      command('tabnew')
+      command('tabnew')
+
+      -- The previous tab is now the third.
+      eq(3, eval('tabpagenr(\'#\')'))
+
+      -- Edit : command line in command-line window
+      feed('q:')
+
+      command_window_id = eval('win_getid()')
+
+      -- At this point switching to the previous tab should have no effect.
+      feed(characters)
+
+      -- Attempting to switch tabs maintains the current window.
+      eq(command_window_id, eval('win_getid()'))
+      eq(completion_visible, eval('complete_info().pum_visible'))
+
+      -- The current tab is still the fourth.
+      eq(4, eval('tabpagenr()'))
+
+      -- The previous tab is still the third.
+      eq(3, eval('tabpagenr(\'#\')'))
+    end
+  end
+  it('does not switch to previous via g<Tab> when editing command line',
```

Comment:
the test descriptions also should be terse where possible, else the logs are harder to read and the tests are harder to refer-to in tickets etc.

Reply:
Fixed.